### PR TITLE
Prometheus: Fix response state

### PR DESCRIPTION
**Pull request by aocenas**
Emit Done only when all requests have finished.

### CODE REVIEW

File: public/app/plugins/datasource/prometheus/datasource.ts [hunk 1]
```diff
@@ -3,9 +3,9 @@ import _ from 'lodash';
 import $ from 'jquery';
 // Services & Utils
 import kbn from 'app/core/utils/kbn';
-import { dateMath, TimeRange, DateTime, AnnotationEvent } from '@grafana/data';
-import { Observable, from, of, merge } from 'rxjs';
-import { filter, map } from 'rxjs/operators';
+import { AnnotationEvent, dateMath, DateTime, LoadingState, TimeRange } from '@grafana/data';
```
Comment (aocenas) [inline public/app/plugins/datasource/prometheus/datasource.ts]: Honestly no idea why this gets ordered all the time. I assume Goland does something when autoimporting stuff need to figure that out.

**Comment by aocenas** [comment]
@torkelo With this fix I think we do not actually need a way to to differentiate between explore and dashboards. Right now the angular Graph panel does not render anything until it gets Done status. This means the rendering is fine.

In react Graph panel, it renders even on Loading state but also waits for Done on first render (so works for image rendering). But as far as I know that one is Alpha so not sure if that needs any change.

So we could implement the sampling to batch the responses a bit as I had in https://github.com/grafana/grafana/pull/19014 but not sure if that is required now.

**Review by hugohaggmark — APPROVED** [review]
LGTM!

**Comment by torkelo** [comment]
Looks good! 
Interesting that the Graph panel waits until Done. Wonder if we can do the same in react (wait until done or streaming state)

**Comment by aocenas** [comment]
> Interesting that the Graph panel waits until Done.

Yes I actually removed it in [my previous PR](https://github.com/grafana/grafana/pull/19014) and aligned it towards how React panel works (which I thought was desired behaviour) and thats why I started fixing seemingly unrelated things there.

**Pull request merged**